src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -37,6 +37,51 @@ export liquid_fraction, phase_partitioning_eq!, saturation_adjustment, phase_par
 # Auxiliary functions, e.g., for diagnostic purposes
 export liquid_ice_pottemp, dry_pottemp, exner
 
+# Thermodynamic states
+export ThermodynamicState, InternalEnergy_Shum_eq, InternalEnergy_Shum_neq
+
+abstract type ThermodynamicState{DT} end
+
+"""
+    InternalEnergy_Shum_eq{DT} <: ThermodynamicState
+
+A thermodynamic state initialized by internal energy
+`e_int`, specific humidity `q_t`,  and density `ρ`,
```

Comment:
@asraero just changed `q_t` and other moisture variables to the more verbose `q_tot`.